spec/controllers/users_controller_spec.rb

```diff
@@ -167,6 +167,20 @@
         post :create, params: { user: valid_attributes }
       }.to change(User, :count).by(1)
     end
+
+    it 'sends a mail confirmation mail to new users' do
+      @request.env["devise.mapping"] = Devise.mappings[:user]
```

Comment:
You should create an rspec module for
`@request.env["devise.mapping"] = Devise.mappings[:user]`

Reply:
how exactly would that look like? Could you provide me a link or do it yourself on this branch?

Reply:
https://github.com/plataformatec/devise/wiki/How-To:-Test-controllers-with-Rails-3-and-4-%28and-RSpec%29

Reply:
Fixed by removing unnecessary calls. This is already done `before :each` in this test, so no duplication.